node/core/backing/src/lib.rs

```diff
@@ -142,8 +146,283 @@ impl ValidatedCandidateCommand {
+					handle_validated_candidate_command(
+						&mut ctx,
+						&mut jobs,
+						relay_parent,
+						command,
+					).await?;
+				} else {
+					panic!("background_validation_tx always alive at this point; qed");
+				}
+			}
+			from_overseer = ctx.recv().fuse() => {
+				match from_overseer? {
+					FromOverseer::Signal(OverseerSignal::ActiveLeaves(update)) => handle_active_leaves_update(
+						&mut ctx,
+						update,
+						&mut jobs,
+						&keystore,
+						&background_validation_tx,
+						&metrics,
+					).await?,
+					FromOverseer::Signal(OverseerSignal::BlockFinalized(..)) => {}
+					FromOverseer::Signal(OverseerSignal::Conclude) => return Ok(()),
+					FromOverseer::Communication { msg } => handle_communication(&mut ctx, &mut jobs, msg).await?,
+				}
+			}
+		)
+	}
+}
+
+async fn handle_validated_candidate_command<Context>(
+	ctx: &mut Context,
+	jobs: &mut HashMap<Hash, JobAndSpan<Context>>,
+	relay_parent: Hash,
+	command: ValidatedCandidateCommand,
+) -> Result<(), Error>
+where
+	Context: SubsystemContext<Message = CandidateBackingMessage>,
+	Context: overseer::SubsystemContext<Message = CandidateBackingMessage>,
+{
+	if let Some(job) = jobs.get_mut(&relay_parent) {
+		job.job.handle_validated_candidate_command(&job.span, ctx, command).await?;
+	} else {
+		// simple race condition; can be ignored - this relay-parent
+		// is no longer relevant.
+	}
+
+	Ok(())
+}
+
+async fn handle_communication<Context>(
+	ctx: &mut Context,
+	jobs: &mut HashMap<Hash, JobAndSpan<Context>>,
+	message: CandidateBackingMessage,
+) -> Result<(), Error>
+where
+	Context: SubsystemContext<Message = CandidateBackingMessage>,
+	Context: overseer::SubsystemContext<Message = CandidateBackingMessage>,
+{
+	match message {
+		CandidateBackingMessage::Second(relay_parent, candidate, pov) => {
+			if let Some(job) = jobs.get_mut(&relay_parent) {
+				job.job.handle_second_msg(&job.span, ctx, candidate, pov).await?;
+			}
+		},
+		CandidateBackingMessage::Statement(relay_parent, statement) => {
+			if let Some(job) = jobs.get_mut(&relay_parent) {
+				job.job.handle_statement_message(&job.span, ctx, statement).await?;
+			}
+		},
+		CandidateBackingMessage::GetBackedCandidates(relay_parent, requested_candidates, tx) =>
+			if let Some(job) = jobs.get_mut(&relay_parent) {
+				job.job.handle_get_backed_candidates_message(requested_candidates, tx)?;
+			},
+	}
+
+	Ok(())
+}
+
+async fn handle_active_leaves_update<Context>(
+	ctx: &mut Context,
+	update: ActiveLeavesUpdate,
+	jobs: &mut HashMap<Hash, JobAndSpan<Context>>,
+	keystore: &SyncCryptoStorePtr,
+	background_validation_tx: &mpsc::Sender<(Hash, ValidatedCandidateCommand)>,
+	metrics: &Metrics,
+) -> Result<(), Error>
+where
+	Context: SubsystemContext<Message = CandidateBackingMessage>,
+	Context: overseer::SubsystemContext<Message = CandidateBackingMessage>,
+{
+	for deactivated in update.deactivated {
+		jobs.remove(&deactivated);
+	}
```

Comment:
This is almost all we need for async backing - just not deactivating stuff quite as quickly.